### PR TITLE
Make the number of hashes in low_power mode configurable.

### PR DESCRIPTION
**Pull request by michael-sewell**
Hi!  If this isn't something you want to accept I won't be offended, but I took heavy inspiration from the great work @shyba is doing on aeon-stak-cpu and this code allows modification of how many hashes to run on each low_power thread. I found it can be useful at times in certain VM situations or other places with lots of L3 cache.  My (somewhat limited) tests show no performance degradation when left at the default 2.

Please LMK if any questions!

### CODE REVIEW

**Comment by psychocrypt** [comment]
@michael-sewell  Thank you for your pull request. We will need some time to evaluate if it is possible to merge this pull request.

There are some questions we need to answer before:
- [ ] is this PR decreasing the hash rate on different systems
- [ ] this change is not compatible with the HWloc #108 and nonHwloc #84 autosuggestion
- [ ] if we add such a power factor we can add this to the default hash function and remove the second implementation (avoid doubling the source code)

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,6 @@ config-debug.txt
 
 # merge original backup files
 *.orig
+
+#Jetbrains IDE project files
+.idea
```
Comment (psychocrypt) [inline .gitignore]: missing EOF

File: crypto/cryptonight_aesni.h [hunk 2]
```diff
@@ -350,108 +351,75 @@ void cryptonight_hash(const void* input, size_t len, void* output, cryptonight_c
...
-			cx = _mm_aesenc_si128(cx, _mm_set_epi64x(axh1, axl1));
+			if (SOFT_AES)
+				cx = soft_aesenc(cx, _mm_set_epi64x(axh[i], axl[i]));
+			else
+				cx = _mm_aesenc_si128(cx, _mm_set_epi64x(axh[i], axl[i]));
 
-		_mm_store_si128((__m128i *)&l1[idx1 & 0x1FFFF0], _mm_xor_si128(bx1, cx));
-		idx1 = _mm_cvtsi128_si64(cx);
-		bx1 = cx;
+			_mm_store_si128((__m128i * ) & l[i][idx[i] & 0x1FFFF0], _mm_xor_si128(bx[i], cx));
```
Comment (psychocrypt) [inline crypto/cryptonight_aesni.h]: A space to much`((__m128i * )` :-)

File: crypto/low_power.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#ifndef XMR_STAK_CPU_LOW_POWER_H
```
Comment (psychocrypt) [inline crypto/low_power.h]: Please use `#pragma once` this is compatible to all compiler and much cleaner than guards per file via precompiler defines.

File: crypto/low_power.h [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+#define XMR_STAK_CPU_LOW_POWER_H
```
Comment (psychocrypt) [inline crypto/low_power.h]: This should exposed to cmake and than optional guarded if someone is not using cmake and missed to set the define.

```
#ifndef XMR_STAK_LOW_POWER_HASHE_COUNT
#   define LOW_POWER_HASHES 2
#else
#   define LOW_POWER_HASHES XMR_STAK_LOW_POWER_HASHE_COUNT
#endif
```

File: .gitignore (hunk 1 above)
Comment (psychocrypt) [inline .gitignore]: missing EOF

**Comment by fireice-uk** [comment]
@michael-sewell Can you please use the C++ template system instead of C defines?

**Comment by michael-sewell** [comment]
All good ideas, will work on these!

**Comment by psychocrypt** [comment]
@michael-sewell  Is there a reason why zou closed this PR?

**Comment by shyba** [comment]
Thank you @michael-sewell for bringing it!

Please don't close it. I think it would be great to have this upstream. :+1: 

If you have any question that you think I may help, feel free to ask.

**Pull request closed without merging**